packages/changed-elements-react/src/widgets/V2Widget/componets/VersionCompareSelectModal.tsx

```diff
@@ -0,0 +1,359 @@
+          namedVersions={result?.namedVersions}
+        />
+      </ModalContent>
+      <ModalButtonBar>
+        <Button
+          styleType="high-visibility"
+          disabled={targetVersion === undefined || result?.namedVersions === undefined}
+          onClick={() => {
+            void _handleOk();
+          }}
+        >
+          {IModelApp.localization.getLocalizedString("VersionCompare:versionCompare.compare")}
+        </Button>
+        <Button onClick={_handleCancel}>
+          {IModelApp.localization.getLocalizedString("UiCore:dialog.cancel")}
+        </Button>
+      </ModalButtonBar>
+    </Modal>
+  );
+}
+
+
+type HandleStartComparisonArgs = {
+  targetVersion: NamedVersion;
+  comparisonJobClient: IComparisonJobClient;
+  result: NamedVersionLoaderResult;
+  iModelConnection: IModelConnection;
+  getDialogOpen: () => boolean;
+};
+
+const handleStartComparison = async (args: HandleStartComparisonArgs) => {
+  if (VersionCompare.manager?.isComparing) {
+    await VersionCompare.manager?.stopComparison();
+  }
+  const currentVersion = args.result.namedVersions.currentVersion?.version;
+  if (args.targetVersion && currentVersion) {
+    runStartComparisonV2({
+      targetVersion: args.targetVersion,
+      comparisonJobClient: args.comparisonJobClient,
+      iModelConnection: args.iModelConnection,
+      currentVersion: currentVersion,
+      getDialogOpen: args.getDialogOpen,
+    }).catch((e) => {
+      Logger.logError(VersionCompare.logCategory, "Could not start version comparison: " + e);
+    });
+  }
+};
+
+type RunStartComparisonV2Args = {
+  targetVersion: NamedVersion;
+  comparisonJobClient: IComparisonJobClient;
+  iModelConnection: IModelConnection;
+  currentVersion: NamedVersion;
+  getDialogOpen: () => boolean;
+};
+
+// todo clean this up once API fix is in. API is currently publishing jobs that are complete but mau not have href
+// this commit has cleaner version 29d2fdaa9cc87ce47a627feb3b5224898107d498
+// while loop may need to be looked at again
+const runStartComparisonV2 = async (args: RunStartComparisonV2Args) => {
+  let { comparisonJob } = await postOrGetComparisonJob({
+    changedElementsClient: args.comparisonJobClient,
+    iTwinId: args.iModelConnection?.iTwinId as string,
+    iModelId: args.iModelConnection?.iModelId as string,
+    startChangesetId: args.targetVersion.changesetId as string,
+    endChangesetId: args.currentVersion.changesetId as string,
+  });
+  if (comparisonJob.status === "Completed" && comparisonJob.comparison && comparisonJob.comparison.href) {
+    void runManagerStartComparisonV2({
+      comparisonJob: { comparisonJob: comparisonJob },
+      comparisonJobClient: args.comparisonJobClient,
+      iModelConnection: args.iModelConnection,
+      targetVersion: args.targetVersion,
+      currentVersion: args.currentVersion,
+    });
+    return;
+  }
+  const toastProgressingInterval = setInterval(() => {
```

Comment:
Please use `try { ... } finally { ... }` statement to clear the interval so that it is easier to make sure it's eventually disposed.